IBPSA/Media/Steam.mo

```diff
@@ -0,0 +1,855 @@
+  algorithm
+    state := ThermodynamicState(
+          p=smoothStep(
+            x,
+            state_a.p,
+            state_b.p,
+            x_small),
+          h=smoothStep(
+            x,
+            state_a.h,
+            state_b.h,
+            x_small),
+          d=density_ph(smoothStep(
+            x,
+            state_a.p,
+            state_b.p,
+            x_small), smoothStep(
+            x,
+            state_a.h,
+            state_b.h,
+            x_small)),
+          T=temperature_ph(smoothStep(
+            x,
+            state_a.p,
+            state_b.p,
+            x_small), smoothStep(
+            x,
+            state_a.h,
+            state_b.h,
+            x_small)));
+    annotation (
+      Inline=true,
+      Documentation(info="<html>
+<p>
+This function is used to approximate the equation
+</p>
+<pre>
+    state = <strong>if</strong> x &gt; 0 <strong>then</strong> state_a <strong>else</strong> state_b;
+</pre>
+<p>
+by a smooth characteristic, so that the expression is continuous and differentiable:
+</p>
+<pre>
+   state := <strong>smooth</strong>(1, <strong>if</strong> x &gt;  x_small <strong>then</strong> state_a <strong>else</strong>
+                      <strong>if</strong> x &lt; -x_small <strong>then</strong> state_b <strong>else</strong> f(state_a, state_b));
+</pre>
+<p>
+This is performed by applying function <strong>Media.Common.smoothStep</strong>(..)
+on every element of the thermodynamic state record.
+</p>
+<p>
+If <strong>mass fractions</strong> X[:] are approximated with this function then this can be performed
+for all <strong>nX</strong> mass fractions, instead of applying it for nX-1 mass fractions and computing
+the last one by the mass fraction constraint sum(X)=1. The reason is that the approximating function has the
+property that sum(state.X) = 1, provided sum(state_a.X) = sum(state_b.X) = 1.
+This can be shown by evaluating the approximating function in the abs(x) &lt; x_small
+region (otherwise state.X is either state_a.X or state_b.X):
+</p>
+<pre>
+    X[1]  = smoothStep(x, X_a[1] , X_b[1] , x_small);
+    X[2]  = smoothStep(x, X_a[2] , X_b[2] , x_small);
+       ...
+    X[nX] = smoothStep(x, X_a[nX], X_b[nX], x_small);
+</pre>
+<p>
+or
+</p>
+<pre>
+    X[1]  = c*(X_a[1]  - X_b[1])  + (X_a[1]  + X_b[1])/2
+    X[2]  = c*(X_a[2]  - X_b[2])  + (X_a[2]  + X_b[2])/2;
+       ...
+    X[nX] = c*(X_a[nX] - X_b[nX]) + (X_a[nX] + X_b[nX])/2;
+    c     = (x/x_small)*((x/x_small)^2 - 3)/4
+</pre>
+<p>
+Summing all mass fractions together results in
+</p>
+<pre>
+    sum(X) = c*(sum(X_a) - sum(X_b)) + (sum(X_a) + sum(X_b))/2
+           = c*(1 - 1) + (1 + 1)/2
+           = 1
+</pre>
+</html>"));
+  end setSmoothState;
+
+  redeclare replaceable function extends saturationState_p
+    "Return saturation property record from pressure"
+  algorithm
+    sat.psat := p;
+    sat.Tsat := saturationTemperature_p(p);
```

Comment:
Better use the function name `saturationTemperature` that takes as an input `p` and returns `T`. Then it has the same API as `Modelica.Media.Interfaces.PartialRealCondensingGases.saturationTemperature`